### PR TITLE
FCS code cleanup

### PR DESCRIPTION
**Pull request by commy2**
replaces: #2093

### CODE REVIEW

**Comment by esteldunedain** [comment]
:+1:

File: addons/fcs/functions/fnc_keyDown.sqf [hunk 1]
```diff
@@ -43,3 +41,5 @@ GVAR(Position) = [
     (getPos _vehicle select 1) + _distance * (_weaponDirection select 1),
     (getPos _vehicle select 2) + _distance * (_weaponDirection select 2)
 ];
```
Comment (esteldunedain) [inline addons/fcs/functions/fnc_keyDown.sqf]: I think this `getPos` should be ASL too

Comment (commy2) [inline addons/fcs/functions/fnc_keyDown.sqf]: noted

File: addons/fcs/functions/fnc_canResetFCS.sqf [hunk 2]
```diff
@@ -1,15 +1,17 @@
...
+ * Return Value:
+ * Boolean <BOOL>
+ *
+ * Public: No
  */
-
 #include "script_component.hpp"
 
-count ((vehicle ACE_player) getVariable [format ["%1_%2", QGVAR(Magazines), [ACE_player] call EFUNC(common,getTurretIndex)], []]) > 1
+params ["_unit"];
```
Comment (commy2) [inline addons/fcs/functions/fnc_canResetFCS.sqf]: .

File: addons/fcs/functions/fnc_onForceUpdate.sqf [hunk 3]
```diff
@@ -1,7 +1,13 @@
 #include "script_component.hpp"
 
-if !([ACE_player, vehicle ACE_player, []] call EFUNC(common,canInteractWith)) exitWith {false};
+params ["_unit"];
```
Comment (commy2) [inline addons/fcs/functions/fnc_onForceUpdate.sqf]: .

**Comment by commy2** [comment]
- holding down the key to adjust for moving vehicles doesn't work at all. (tank fires straight north)
- pg up and pg dont work (they call keyUp ?)

**Comment by commy2** [comment]
Done. Merge?

**Comment by bux578** [comment]
For 3.4.2? I would vote "yes". (but target "release")

**Pull request closed without merging**